src/net.h

```diff
@@ -114,12 +114,12 @@ struct CSerializedNetMsg
 };
 
 enum class ConnectionType {
-    INBOUND,
-    OUTBOUND,
-    MANUAL,
-    FEELER,
-    BLOCK_RELAY,
-    ADDR_FETCH,
+    INBOUND, /**< peer initiated connections */
```

Comment:
It's great to see documentation around types. I think you should go further and describe exactly the set of assumptions we are doing for each type (like outbound more trusted than inbound, outbound favored for relay, block_relay_only to avoid leaking the full of our `block`-topology ...) and design goals of more obscure ones (feeler to sanitize addr db, addr fetch to boostrap our view of the network). With `ConnectionType` you can even grep code locations and points directly to where we are taking decisions based on type. That would be awesome for newcomers :)

As a follow-up ofc.